[
   {
      "body" : "Just to clarify: that address is yours (ie no money was lost, only isnt being displayed by a normal \"getbalance\" rpc call (ie getunconfirmedbalance is correct)?",
      "created_at" : "2017-01-24T02:39:14Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274684906",
      "id" : 274684906,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T02:39:14Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274684906",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/649246?v=3",
         "events_url" : "https://api.github.com/users/TheBlueMatt/events{/privacy}",
         "followers_url" : "https://api.github.com/users/TheBlueMatt/followers",
         "following_url" : "https://api.github.com/users/TheBlueMatt/following{/other_user}",
         "gists_url" : "https://api.github.com/users/TheBlueMatt/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/TheBlueMatt",
         "id" : 649246,
         "login" : "TheBlueMatt",
         "organizations_url" : "https://api.github.com/users/TheBlueMatt/orgs",
         "received_events_url" : "https://api.github.com/users/TheBlueMatt/received_events",
         "repos_url" : "https://api.github.com/users/TheBlueMatt/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/TheBlueMatt/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/TheBlueMatt/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/TheBlueMatt"
      }
   },
   {
      "body" : "The money isn't showing in either `getunconfirmedbalance` or `getbalance`. I'll let you know what happens after it confirms, but I strongly suspect it'll arrive back in my (visible) wallet. I have the private key for the address it's in (I double checked that :P), so there's absolutely no risk in that sense.\r\n\r\nI've never had a problem with the wallet and unconfirmed change, but it seems to have been triggered by the spending of spending unconfirmed change, if I had to guess.\r\n\r\n\r\nBut my big concern is, what would happen if the transaction chain never confirmed? Seems like it's possible someone could \"lose\" money this way",
      "created_at" : "2017-01-24T04:04:15Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274704309",
      "id" : 274704309,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T04:12:29Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274704309",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "Also not sure if it's relevant, but sometimes I send money to myself.  (because sometimes my users make on-chain payments to each other). So in that case, both the change output and the other output are owned by my wallet.\r\n\r\nThe (sole) input that f600e0... is sourcing is one of those cases. It sources  702a1bfe235405069e50340c10e0b8236e25124ab5bab89a87ab3859a4f97f3b:0  which is owned by my wallet (non change ouput). And 702a1bfe235405069e50340c10e0b8236e25124ab5bab89a87ab3859a4f97f3b:1 was the change in that case.\r\n\r\nNot sure if it's relevant, I suspect not, but just throwing that out there. I strongly suspect the issue was triggered by 702a1bfe235405069e50340c10e0b8236e25124ab5bab89a87ab3859a4f97f3b when it sourced a bunch of unconfirmed change outputs, (who also might have sourced unconfirmed change outputs?)",
      "created_at" : "2017-01-24T04:32:21Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274707290",
      "id" : 274707290,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T04:33:44Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274707290",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "I assume you are running 0.13.2, right?\n",
      "created_at" : "2017-01-24T14:15:42Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274813456",
      "id" : 274813456,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T14:15:42Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274813456",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/6399679?v=3",
         "events_url" : "https://api.github.com/users/MarcoFalke/events{/privacy}",
         "followers_url" : "https://api.github.com/users/MarcoFalke/followers",
         "following_url" : "https://api.github.com/users/MarcoFalke/following{/other_user}",
         "gists_url" : "https://api.github.com/users/MarcoFalke/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/MarcoFalke",
         "id" : 6399679,
         "login" : "MarcoFalke",
         "organizations_url" : "https://api.github.com/users/MarcoFalke/orgs",
         "received_events_url" : "https://api.github.com/users/MarcoFalke/received_events",
         "repos_url" : "https://api.github.com/users/MarcoFalke/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/MarcoFalke/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/MarcoFalke/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/MarcoFalke"
      }
   },
   {
      "body" : "Yeah.\r\nAlso the problem is much worse than I thought, that transaction has confirmed and the funds are no where to be seen.\r\n\r\n$ bitcoin-cli listunspent 0 | grep f600e0a64f2166deb03804499c09e1b3c07636a2d5725f4a10852774da02dd8a\r\n\r\nreturn nothing as well. However, that change address is actually in my wallet.",
      "created_at" : "2017-01-24T14:23:38Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274815741",
      "id" : 274815741,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T14:24:03Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274815741",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "Well the good news is that I ran `bitcoind -rescan` and it found my money =)",
      "created_at" : "2017-01-24T15:36:02Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274838662",
      "id" : 274838662,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T15:36:02Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274838662",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "Can you elaborate on what kinds of action you've done with this wallet? (RPCs, GUI operations, copying files around...). Balances disappearing is very serious, and we should figure out what caused this.",
      "created_at" : "2017-01-24T15:39:08Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274839625",
      "id" : 274839625,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T15:39:18Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274839625",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/548488?v=3",
         "events_url" : "https://api.github.com/users/sipa/events{/privacy}",
         "followers_url" : "https://api.github.com/users/sipa/followers",
         "following_url" : "https://api.github.com/users/sipa/following{/other_user}",
         "gists_url" : "https://api.github.com/users/sipa/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/sipa",
         "id" : 548488,
         "login" : "sipa",
         "organizations_url" : "https://api.github.com/users/sipa/orgs",
         "received_events_url" : "https://api.github.com/users/sipa/received_events",
         "repos_url" : "https://api.github.com/users/sipa/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/sipa/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/sipa/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/sipa"
      }
   },
   {
      "body" : "I don't have the gui installed, so I don't do anything weird with that. I only interact with it over rpc, 99% of the time itor do anything weird. But I use it as a backend for my service and it gets hit by a lot of rpc calls to `sendtoaddress`.\r\n\r\nAs far as I know, this is the first time it's ever really been forced to spend from a chain of unconfirmed transactions -- so I believe this is what must have triggered it.\r\n\r\nNot sure if it matters, but the wallet itself is a few years old (so it's non-hd) and I have `txindex=1` set",
      "created_at" : "2017-01-24T15:40:31Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274840035",
      "id" : 274840035,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T15:43:56Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274840035",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "I'm also using totally default settings regarding spending unconfirmed funds.\r\n\r\n\r\nI suspect it's related to this transaction: 702a1bfe235405069e50340c10e0b8236e25124ab5bab89a87ab3859a4f97f3b\r\n\r\nWhich was generated via rpc `sendtoaddress 1JyZ2dJq2tz6RDTuNQfwg498Qu2CoCp7TG 66.6` where the interesting thing is that `1JyZ2dJq2tz6RDTuNQfwg498Qu2CoCp7TG` is actually part of my wallet. It's also worth noting that in order to create that transaction, it needed to source a bunch of unconfirmed transactions itself.\r\n\r\nThe next transaction was generated via rpc `sendtoaddress 1GUJWXTG9bCL72G2nPjEaWpSLC9FG5vfF2 0.0997`  actually sourced that unconfirmed money from the \"main\" output of 702a1bfe235405069e50340c10e0b8236e25124ab5bab89a87ab3859a4f97f3b  (I guess because it trusts it, because it comes from me, even though it's not a change output?).\r\n\r\nAnd it was the change from _that_ send (that arrived in 19pozzqPQ6tmubTPGY5KXy3GR5agAaAedA) which appeared to have been permanently lost until I rescanned the wallet.\r\n\r\n---\r\n\r\nIt also seems like a pretty rare case. My wallet has made 126,591 sends and this appears to be the only case of it. I was kind of hoping it happened more often, and found all these lost riches after I did the rescan :(",
      "created_at" : "2017-01-24T15:55:06Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274844632",
      "id" : 274844632,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T16:09:50Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274844632",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "I know you've already said this, but to just be 100% explicit, which getbalance call were you using?\r\n\r\n`getbalance` ?\r\n\r\nnot `getbalance '*'` or `getbalance ''`?\r\n\r\n",
      "created_at" : "2017-01-24T15:59:28Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274845980",
      "id" : 274845980,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T15:59:28Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274845980",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/4360349?v=3",
         "events_url" : "https://api.github.com/users/morcos/events{/privacy}",
         "followers_url" : "https://api.github.com/users/morcos/followers",
         "following_url" : "https://api.github.com/users/morcos/following{/other_user}",
         "gists_url" : "https://api.github.com/users/morcos/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/morcos",
         "id" : 4360349,
         "login" : "morcos",
         "organizations_url" : "https://api.github.com/users/morcos/orgs",
         "received_events_url" : "https://api.github.com/users/morcos/received_events",
         "repos_url" : "https://api.github.com/users/morcos/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/morcos/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/morcos/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/morcos"
      }
   },
   {
      "body" : "Yeah, I was using just normal `bitcoin-cli getbalance`. The actual reason I noticed was that my site couldn't process any more withdrawals, even though the wallet should've had lots of funds in it. (I don't programmatically query the balance, I just blindly fire off `sendtoaddress` rpc requests)\r\n\r\nWhen doing `bitcoin-cli listunspent 0` it also didn't show the output, both before and after the transaction had confirmed. (Although after the -rescan, it did)",
      "created_at" : "2017-01-24T16:00:46Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274846412",
      "id" : 274846412,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T16:06:41Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274846412",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "Could you check your debug.log for \r\n`AddToWallet f600e0a64f2166deb03804499c09e1b3c07636a2d5725f4a10852774da02dd8a`?\r\n\r\nIt should have appeared when you placed the transaction, again when that transaction was confirmed in a block, and again when you did the rescan and it got to that block.\r\n\r\n\r\n\r\n",
      "created_at" : "2017-01-24T16:26:54Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274854915",
      "id" : 274854915,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T16:26:54Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274854915",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/4360349?v=3",
         "events_url" : "https://api.github.com/users/morcos/events{/privacy}",
         "followers_url" : "https://api.github.com/users/morcos/followers",
         "following_url" : "https://api.github.com/users/morcos/following{/other_user}",
         "gists_url" : "https://api.github.com/users/morcos/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/morcos",
         "id" : 4360349,
         "login" : "morcos",
         "organizations_url" : "https://api.github.com/users/morcos/orgs",
         "received_events_url" : "https://api.github.com/users/morcos/received_events",
         "repos_url" : "https://api.github.com/users/morcos/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/morcos/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/morcos/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/morcos"
      }
   },
   {
      "body" : "```\r\n$ cat ~/.bitcoin/debug.log | grep f600e0a64f2166deb03804499c09e1b3c07636a2d5725f4a10852774da02dd8a\r\n```\r\nOnly returns one entry:\r\n```\r\n2017-01-24 15:21:24 AddToWallet f600e0a64f2166deb03804499c09e1b3c07636a2d5725f4a10852774da02dd8a\r\n```\r\nfrom the time of rescanning.\r\n\r\n\r\nAlthough the first entry I see in my ~/.bitcoin/debug.log is very recent:\r\n2017-01-24 13:38:51 receive version message: /Bither1.6.0/: version 70001, blocks=449795, us=127.0.0.1:8333, peer=625536\r\n\r\n\r\nSeems like it's automatically truncated?",
      "created_at" : "2017-01-24T16:31:07Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274856464",
      "id" : 274856464,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T16:36:12Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274856464",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "This however is quite interesting:\r\n\r\n```\r\n$ cat ~/.bitcoin/debug.log | grep 702a1bfe235405069e50340c10e0b8236e25124ab5bab89a87ab3859a4f97f3b\r\n2017-01-24 15:21:23 AddToWallet 702a1bfe235405069e50340c10e0b8236e25124ab5bab89a87ab3859a4f97f3b  \r\n2017-01-24 15:21:23 Transaction e1a732892538809478e344bf47c513e41c5dc0eb33fd38d8d54695c79b3d9796 (in block 00000000000000000165a1a4b244278015d3d538d503609f2301199a7a3cea72) conflicts with wallet transaction 117f591dc7e0ea1779e415d3f0d9dffe14004827fe869279d7a92723551fe53c (both spend 702a1bfe235405069e50340c10e0b8236e25124ab5bab89a87ab3859a4f97f3b:1)\r\n```\r\n\r\n\r\nSo transaction `e1a732892538809478e344bf47c513e41c5dc0eb33fd38d8d54695c79b3d9796` on the blockchain is a quick-and-dirty transaction I created (manually) and broadcasted myself to do some CPFP action.  While doing that, it seems to have conflicted with a transaction my wallet generated: `117f591dc7e0ea1779e415d3f0d9dffe14004827fe869279d7a92723551fe53c` whose contents are:\r\n\r\n```\r\n010000000488083f430529034ff9418ff04b612bef176d327de22f3bb7e6625d2235587e5e000000006b483045022100f149050241f1fb1f54550471adf1affcd57e4cbb36e4148574a29a457f78687402200be84b23455077c9b9dac326c6017ea12d471ff63b4e6e8d07bdd63e3304d4b7012103b0af308aa07c4dbdb5ac7997588ba65335a103914aa2e390100d3bfa8d3da2aafeffffff887326bd759dee8a863767a03851c3edfc099dc9909c602d159a32de77e563fd000000006946304302206383b94705c164ffab60b7e24b27934d05cef017d8a607c553cb5e963ac1379b021f21d405918a8b7655f6ff3da6f37a34cdcf128678cd886c2b353352e10c6d3e012102cd3a1be957907775d65ef28cf57dd4cede822a50ac2dca03941f3565e22ef848feffffffb23889605c2ba89b378374e0e3a59326829e43448bfe75d68e91640c74b9d8ce010000006b483045022100955b7904ba49c12a32b832fd8a56bb92697c621d02003ccb0a8b46ff5455527f02200450eebfe6ff50e10e29cdf7d4a95da2e0547001233b70d97f1be3efa0d82d5c0121031ba7139f7c39f37f4c352c8b5cc25348e55cb7239e9a9b1fc1b4218e4970345dfeffffff3b7ff9a45938ab879ab8bab54a12256e23b8e0100c34509e06055423fe1b2a70010000006a473044022038d0d9d6bbe27a752130021efe6dfe72ab8da8818668f98a1d6086210cdc1f5f02207f9658ad2bb99d3bbf7e5854a1fdb986b55b9cdbd9fe35a959ff9e1f88fd811401210210b958b5f9c4c0ce64e0e03afb96d7eae2d53d4a534daf174ed8de766afc26e0feffffff02eb2e0000000000001976a9144a1e7cf8cee728aa3cacca6a4e02f5bafc9d3f6d88acd05e3000000000001976a9149925601955f59f337a20b752c2e93225683e878a88aca8dc0600\r\n```\r\n\r\nIt seems quite possible this was somehow involved in me losing funds\r\n\r\n",
      "created_at" : "2017-01-24T16:36:50Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274858497",
      "id" : 274858497,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T16:48:33Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274858497",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "It's a shame I don't have better logs for it, when I excecuted the RPC command\r\n`sendtoaddress 1ExmAEmkdQcMxHQW7a37TojVR7hxhaaxSK 0.0317` my application logged it as failing (although unfortunately I didn't log the exact error lol).\r\n\r\nHowever, there's a wallet transaction above that was generated (117f591dc7e0ea1779e415d3f0d9dffe14004827fe869279d7a92723551fe53c) for it. However, none of the block explorers seem to have ever seen that transaction, so it was likely never pushed onto the network. (probably cause of the conflicting manual CPFP transaction i made: e1a732892538809478e344bf47c513e41c5dc0eb33fd38d8d54695c79b3d9796 ) ",
      "created_at" : "2017-01-24T17:02:59Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274867746",
      "id" : 274867746,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T17:02:59Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274867746",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "your debug log doesn't include the times when you actually placed the transactions right?\r\nI saw them relayed around `2017-01-23 23:27:43`, does your log have anything from that time period?",
      "created_at" : "2017-01-24T17:10:13Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274869989",
      "id" : 274869989,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T17:10:13Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274869989",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/4360349?v=3",
         "events_url" : "https://api.github.com/users/morcos/events{/privacy}",
         "followers_url" : "https://api.github.com/users/morcos/followers",
         "following_url" : "https://api.github.com/users/morcos/following{/other_user}",
         "gists_url" : "https://api.github.com/users/morcos/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/morcos",
         "id" : 4360349,
         "login" : "morcos",
         "organizations_url" : "https://api.github.com/users/morcos/orgs",
         "received_events_url" : "https://api.github.com/users/morcos/received_events",
         "repos_url" : "https://api.github.com/users/morcos/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/morcos/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/morcos/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/morcos"
      }
   },
   {
      "body" : "Nope :(\r\n\r\nThe earliest entry in my ~/.bitcoin/debug.log is from `2017-01-24 13:38:51`. I'm not sure why that is, I certainly didn't manually clear it or anything",
      "created_at" : "2017-01-24T17:12:35Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274870690",
      "id" : 274870690,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T17:12:35Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274870690",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "The debug.log file gets shrunk on restart.  See #9625.\r\n\r\nI'm having trouble knowing where to look next for what might have caused this problem.\r\n\r\n- Were there any other manual double spends or other manual transactions that you created?\r\n- I assume you had `-spendzeroconfchange` set to true (the default)?\r\n- Did you abandon any transactions?\r\n- Were any of these transactions \"mixed-debit\", that is spent some of your inputs and some of somebody elses?\r\n- Do you know if the transaction was in your node's mempool?\r\n\r\nWith the information you've given, I can't see why your wallet wouldn't have recognized the output of f600e0a64f2166deb03804499c09e1b3c07636a2d5725f4a10852774da02dd8a as your money.\r\n\r\nIf this happens again (hopefully not), then please make a copy of the debug.log before restarting your node.\r\n\r\n",
      "created_at" : "2017-01-24T17:55:38Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274882828",
      "id" : 274882828,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T17:55:38Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274882828",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/4360349?v=3",
         "events_url" : "https://api.github.com/users/morcos/events{/privacy}",
         "followers_url" : "https://api.github.com/users/morcos/followers",
         "following_url" : "https://api.github.com/users/morcos/following{/other_user}",
         "gists_url" : "https://api.github.com/users/morcos/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/morcos",
         "id" : 4360349,
         "login" : "morcos",
         "organizations_url" : "https://api.github.com/users/morcos/orgs",
         "received_events_url" : "https://api.github.com/users/morcos/received_events",
         "repos_url" : "https://api.github.com/users/morcos/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/morcos/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/morcos/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/morcos"
      }
   },
   {
      "body" : "> Were there any other manual double spends or other manual transactions that you created?\r\n\r\nNo, e1a732892538809478e344bf47c513e41c5dc0eb33fd38d8d54695c79b3d9796 was the only transaction I have manually created in a while. It wasn't intended to be a double-spend, but to child-pays for parent. Guess I had bad luck with the timing of that.\r\n\r\n> I assume you had -spendzeroconfchange set to true (the default)?\r\n\r\nYeah, I'm running with default settings\r\n\r\n> Did you abandon any transactions?\r\n\r\nNo, I've never used abandon\r\n\r\n> Were any of these transactions \"mixed-debit\", that is spent some of your inputs and some of somebody elses?\r\n\r\nNo, all transactions (with the exception of the single CPFP) are simply created with `sendtoaddress $addr $amount` and all inputs are owned by me\r\n\r\n> Do you know if the transaction was in your node's mempool?\r\n\r\nI forgot to check, but even after it confirmed it never showed up\r\n\r\n> If this happens again (hopefully not), then please make a copy of the debug.log before restarting your node.\r\n\r\nSure, dumb of me to not do so",
      "created_at" : "2017-01-24T18:01:56Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274884586",
      "id" : 274884586,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T18:03:49Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274884586",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "@RHavar Ok I think I have an idea\r\nCan you run `gettransaction 117f591dc7e0ea1779e415d3f0d9dffe14004827fe869279d7a92723551fe53c`\r\n\r\nI'm guessing that the `117` transaction might have spent the `f60` output, causing it to appear spent, but since `117` wasn't in your mempool, you didn't see the output of that as yours.\r\n\r\nThat situation should have rectified itself when `117` was marked conflicted though, which would have happened when `e1a` was confirmed in block 449727?  But that seems not to have happened?   \r\n\r\n",
      "created_at" : "2017-01-24T18:18:09Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274889188",
      "id" : 274889188,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T18:18:09Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274889188",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/4360349?v=3",
         "events_url" : "https://api.github.com/users/morcos/events{/privacy}",
         "followers_url" : "https://api.github.com/users/morcos/followers",
         "following_url" : "https://api.github.com/users/morcos/following{/other_user}",
         "gists_url" : "https://api.github.com/users/morcos/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/morcos",
         "id" : 4360349,
         "login" : "morcos",
         "organizations_url" : "https://api.github.com/users/morcos/orgs",
         "received_events_url" : "https://api.github.com/users/morcos/received_events",
         "repos_url" : "https://api.github.com/users/morcos/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/morcos/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/morcos/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/morcos"
      }
   },
   {
      "body" : "`bitcoin-cli gettransaction 117f591dc7e0ea1779e415d3f0d9dffe14004827fe869279d7a92723551fe53c`\r\n\r\n```\r\n{\r\n  \"amount\": -0.03170000,\r\n  \"fee\": -0.00046706,\r\n  \"confirmations\": -98,\r\n  \"trusted\": false,\r\n  \"txid\": \"117f591dc7e0ea1779e415d3f0d9dffe14004827fe869279d7a92723551fe53c\",\r\n  \"walletconflicts\": [\r\n    \"044fc10bc12fe9d195590e76242c4b9474d7ade638e69238b0fb4c3734c0e173\", \r\n    \"e1a732892538809478e344bf47c513e41c5dc0eb33fd38d8d54695c79b3d9796\", \r\n    \"5cda8994d62263c536649948d333197ff8e3e6d043b983388ac0fe5ee49b3dc2\"\r\n  ],\r\n  \"time\": 1485214577,\r\n  \"timereceived\": 1485214577,\r\n  \"bip125-replaceable\": \"unknown\",\r\n  \"details\": [\r\n    {\r\n      \"account\": \"\",\r\n      \"address\": \"1ExmAEmkdQcMxHQW7a37TojVR7hxhaaxSK\",\r\n      \"category\": \"send\",\r\n      \"amount\": -0.03170000,\r\n      \"vout\": 1,\r\n      \"fee\": -0.00046706,\r\n      \"abandoned\": false\r\n    }\r\n  ],\r\n  \"hex\": \"010000000488083f430529034ff9418ff04b612bef176d327de22f3bb7e6625d2235587e5e000000006b483045022100f149050241f1fb1f54550471adf1affcd57e4cbb36e4148574a29a457f78687402200be84b23455077c9b9dac326c6017ea12d471ff63b4e6e8d07bdd63e3304d4b7012103b0af308aa07c4dbdb5ac7997588ba65335a103914aa2e390100d3bfa8d3da2aafeffffff887326bd759dee8a863767a03851c3edfc099dc9909c602d159a32de77e563fd000000006946304302206383b94705c164ffab60b7e24b27934d05cef017d8a607c553cb5e963ac1379b021f21d405918a8b7655f6ff3da6f37a34cdcf128678cd886c2b353352e10c6d3e012102cd3a1be957907775d65ef28cf57dd4cede822a50ac2dca03941f3565e22ef848feffffffb23889605c2ba89b378374e0e3a59326829e43448bfe75d68e91640c74b9d8ce010000006b483045022100955b7904ba49c12a32b832fd8a56bb92697c621d02003ccb0a8b46ff5455527f02200450eebfe6ff50e10e29cdf7d4a95da2e0547001233b70d97f1be3efa0d82d5c0121031ba7139f7c39f37f4c352c8b5cc25348e55cb7239e9a9b1fc1b4218e4970345dfeffffff3b7ff9a45938ab879ab8bab54a12256e23b8e0100c34509e06055423fe1b2a70010000006a473044022038d0d9d6bbe27a752130021efe6dfe72ab8da8818668f98a1d6086210cdc1f5f02207f9658ad2bb99d3bbf7e5854a1fdb986b55b9cdbd9fe35a959ff9e1f88fd811401210210b958b5f9c4c0ce64e0e03afb96d7eae2d53d4a534daf174ed8de766afc26e0feffffff02eb2e0000000000001976a9144a1e7cf8cee728aa3cacca6a4e02f5bafc9d3f6d88acd05e3000000000001976a9149925601955f59f337a20b752c2e93225683e878a88aca8dc0600\"\r\n}\r\n```",
      "created_at" : "2017-01-24T18:21:03Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274890020",
      "id" : 274890020,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T18:21:03Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274890020",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "well that idea wasn't quite right, sorry.\r\n\r\ndo you know if there were any other transactions that didn't work like the `117` one?\r\n",
      "created_at" : "2017-01-24T18:32:39Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274893351",
      "id" : 274893351,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T18:32:39Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274893351",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/4360349?v=3",
         "events_url" : "https://api.github.com/users/morcos/events{/privacy}",
         "followers_url" : "https://api.github.com/users/morcos/followers",
         "following_url" : "https://api.github.com/users/morcos/following{/other_user}",
         "gists_url" : "https://api.github.com/users/morcos/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/morcos",
         "id" : 4360349,
         "login" : "morcos",
         "organizations_url" : "https://api.github.com/users/morcos/orgs",
         "received_events_url" : "https://api.github.com/users/morcos/received_events",
         "repos_url" : "https://api.github.com/users/morcos/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/morcos/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/morcos/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/morcos"
      }
   },
   {
      "body" : "or any other `conflicts with wallet transaction` in your debug.log?",
      "created_at" : "2017-01-24T18:34:30Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274893870",
      "id" : 274893870,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T18:34:30Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274893870",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/4360349?v=3",
         "events_url" : "https://api.github.com/users/morcos/events{/privacy}",
         "followers_url" : "https://api.github.com/users/morcos/followers",
         "following_url" : "https://api.github.com/users/morcos/following{/other_user}",
         "gists_url" : "https://api.github.com/users/morcos/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/morcos",
         "id" : 4360349,
         "login" : "morcos",
         "organizations_url" : "https://api.github.com/users/morcos/orgs",
         "received_events_url" : "https://api.github.com/users/morcos/received_events",
         "repos_url" : "https://api.github.com/users/morcos/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/morcos/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/morcos/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/morcos"
      }
   },
   {
      "body" : "There's quite a few, but none recently. Here's the tail of it:\r\n\r\n```\r\n2017-01-24 15:20:14 Transaction 4e051ad4f11de9d2d18789a1be3bd042ab580f39742322f65e4440a8e755d02a (in block 000000000000000000f3a5d13ecb2e75863b16a12f2a8505bcf5539a8c1d3a2c) conflicts with wallet transaction e5db9b3de5e8b4d6c1ab73c538629d8e159c310c2df7600f6610eabd0b643793 (both spend b3c26fca456ef4c4003b7fa34e4b629ba457be7899c33552b6240af771b81bce:100)\r\n2017-01-24 15:21:23 Transaction e1a732892538809478e344bf47c513e41c5dc0eb33fd38d8d54695c79b3d9796 (in block 00000000000000000165a1a4b244278015d3d538d503609f2301199a7a3cea72) conflicts with wallet transaction 117f591dc7e0ea1779e415d3f0d9dffe14004827fe869279d7a92723551fe53c (both spend 702a1bfe235405069e50340c10e0b8236e25124ab5bab89a87ab3859a4f97f3b:1)\r\n2017-01-24 15:21:24 Transaction 044fc10bc12fe9d195590e76242c4b9474d7ade638e69238b0fb4c3734c0e173 (in block 000000000000000001784ecbd952443a6c94880c8468dda2faf0590e6c4600fa) conflicts with wallet transaction 117f591dc7e0ea1779e415d3f0d9dffe14004827fe869279d7a92723551fe53c (both spend 5e7e5835225d62e6b73b2fe27d326d17ef2b614bf08f41f94f032905433f0888:0)\r\n2017-01-24 15:21:24 Transaction 044fc10bc12fe9d195590e76242c4b9474d7ade638e69238b0fb4c3734c0e173 (in block 000000000000000001784ecbd952443a6c94880c8468dda2faf0590e6c4600fa) conflicts with wallet transaction 117f591dc7e0ea1779e415d3f0d9dffe14004827fe869279d7a92723551fe53c (both spend ced8b9740c64918ed675fe8b44439e822693a5e3e07483379ba82b5c608938b2:1)\r\n2017-01-24 17:45:30 Transaction 5cda8994d62263c536649948d333197ff8e3e6d043b983388ac0fe5ee49b3dc2 (in block 0000000000000000017c5ccc3943c9e63da260d6ee059a736fb78317aee2a70d) conflicts with wallet transaction 117f591dc7e0ea1779e415d3f0d9dffe14004827fe869279d7a92723551fe53c (both spend fd63e577de329a152d609c90c99d09fcedc35138a06737868aee9d75bd267388:0)\r\n```\r\n\r\n(the first one in that tail, is from 18 days ago)",
      "created_at" : "2017-01-24T18:43:01Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274896288",
      "id" : 274896288,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T18:43:01Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274896288",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "OK..  Thanks for your help with this. \r\nI'm not sure how much you want to keep answering questions, but here you go.\r\n\r\nYou said you are running 0.13.2 right?  Because it's pretty unlikely for sendtoaddress to fail with 0.13.2, it is possible that the wallet txid it returns didn't get accepted to the mempool, but it doesn't look like a failure.  Do you know if your application failing on `117` was a result of the return value of sendtoaddress or possibly some other logic, like looking for your tx in the mempool?\r\n\r\nThe other thing that's worth thinking about is the tx `6ff221aee6174acc2263c64b1deec94ddfa1ad0b2877b30ccd580476f08ab7e6` which  spends that 66.5 BTC output from the `f60` transaction.  Can you try searching for that in your log and also do a gettransaction on it (in the off chance it has any wallet conflicts)?  If you had generated that one some time ago but it had failed to be accepted to your mempool (in the same way `117` was) then it's possible it would have looked like your money was missing.  But again with 0.13.2, the `6ff` tx should have been continuously tried to be reaccepted to your mempool, so you should not have needed to wait for a block in order to see that money as yours again.\r\n",
      "created_at" : "2017-01-24T19:08:00Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274903554",
      "id" : 274903554,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T19:08:00Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274903554",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/4360349?v=3",
         "events_url" : "https://api.github.com/users/morcos/events{/privacy}",
         "followers_url" : "https://api.github.com/users/morcos/followers",
         "following_url" : "https://api.github.com/users/morcos/following{/other_user}",
         "gists_url" : "https://api.github.com/users/morcos/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/morcos",
         "id" : 4360349,
         "login" : "morcos",
         "organizations_url" : "https://api.github.com/users/morcos/orgs",
         "received_events_url" : "https://api.github.com/users/morcos/received_events",
         "repos_url" : "https://api.github.com/users/morcos/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/morcos/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/morcos/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/morcos"
      }
   },
   {
      "body" : "I'm also curious about `6ff221aee6174acc2263c64b1deec94ddfa1ad0b2877b30ccd580476f08ab7e6`. If it somehow hadnt made it into your mempool (do you have a super small maxmempool or so?) then that would explain most of this, I'm just not clear on how that would have happened.",
      "created_at" : "2017-01-24T19:21:48Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274908196",
      "id" : 274908196,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T19:21:48Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274908196",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/649246?v=3",
         "events_url" : "https://api.github.com/users/TheBlueMatt/events{/privacy}",
         "followers_url" : "https://api.github.com/users/TheBlueMatt/followers",
         "following_url" : "https://api.github.com/users/TheBlueMatt/following{/other_user}",
         "gists_url" : "https://api.github.com/users/TheBlueMatt/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/TheBlueMatt",
         "id" : 649246,
         "login" : "TheBlueMatt",
         "organizations_url" : "https://api.github.com/users/TheBlueMatt/orgs",
         "received_events_url" : "https://api.github.com/users/TheBlueMatt/received_events",
         "repos_url" : "https://api.github.com/users/TheBlueMatt/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/TheBlueMatt/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/TheBlueMatt/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/TheBlueMatt"
      }
   },
   {
      "body" : "> You said you are running 0.13.2 right? Because it's pretty unlikely for sendtoaddress to fail with 0.13.2, it is possible that the wallet txid it returns didn't get accepted to the mempool, but it doesn't look like a failure.\r\n\r\nYeah, I'm running 0.13.2. Unfortunately, I don't log the error, just the fact it failed. So I have no idea why it failed =/. It's even possible it took longer than usual and timed out. Send to address normally takes 5-20 seconds to return, so it could have conceivably hit the 30 second timeout.\r\n\r\n> The other thing that's worth thinking about is the tx 6ff221aee6174acc2263c64b1deec94ddfa1ad0b2877b30ccd580476f08ab7e6 which spends that 66.5 BTC output from the f60 transaction. Can you try searching for that in your log and also do a gettransaction on it (in the off chance it has any wallet conflicts)\r\n\r\n\r\n`cat ~/.bitcoin/debug.log | grep 6ff221aee6174acc2263c64b1deec94ddfa1ad0b2877b30ccd580476f08ab7e6 -A 5 -B 5`\r\n\r\ngives:\r\n\r\n```\r\n$ cat ~/.bitcoin/debug.log | grep 6ff221aee6174acc2263c64b1deec94ddfa1ad0b2877b30ccd580476f08ab7e6 -A 5 -B 5\r\n2017-01-24 15:21:25 AddToWallet ffcd29a98909b49d040be27e9fb9873a5970317e75567dce3efc701b97bd5d54  \r\n2017-01-24 15:21:25 AddToWallet ba10a850c5c674b3a13a6d1914d7c0a8e05ea2f8ce9eb7b02726b003ef83d6c1  \r\n2017-01-24 15:21:26 AddToWallet b7c6dae15800a2f35ac7cbac854dbb95700b9654b09a3429684b071436bc21c8  \r\n2017-01-24 15:21:26 AddToWallet e01a57d406efc73fa8737fc59e19992997f53850793b50c2856ca01d01f2cddf  \r\n2017-01-24 15:21:26 AddToWallet 94afa233b9b924eb82d37900754770a671e6f584237fd56c0ee135c434bca94b  \r\n2017-01-24 15:21:26 AddToWallet 6ff221aee6174acc2263c64b1deec94ddfa1ad0b2877b30ccd580476f08ab7e6  \r\n2017-01-24 15:21:26 AddToWallet f43459559a1467dce61266de8c9ff24301f5d47458936f5ce297b3e570781a42  \r\n2017-01-24 15:21:26 AddToWallet fd3b229bd87357ceb521f8657104f0954563d2d80340145a6eeb1766821b1e01  \r\n2017-01-24 15:21:26 AddToWallet a6965943dce6112cdabe9a9e36d39f432b419054dfc893487b5224d7322b61a7  \r\n2017-01-24 15:21:26 AddToWallet 3bc722f7b210a80efa5a8de030f64ab8ce1940a5e45801b57d9444dbbb99e67c  \r\n2017-01-24 15:21:26 AddToWallet 46d82330d53afa5087470118cdf4cf69771aed94851bd9282f107cd383f7676c  \r\n--\r\n2017-01-24 15:32:09 Relaying wtx ffcd29a98909b49d040be27e9fb9873a5970317e75567dce3efc701b97bd5d54\r\n2017-01-24 15:32:09 Relaying wtx ba10a850c5c674b3a13a6d1914d7c0a8e05ea2f8ce9eb7b02726b003ef83d6c1\r\n2017-01-24 15:32:09 Relaying wtx b7c6dae15800a2f35ac7cbac854dbb95700b9654b09a3429684b071436bc21c8\r\n2017-01-24 15:32:09 Relaying wtx e01a57d406efc73fa8737fc59e19992997f53850793b50c2856ca01d01f2cddf\r\n2017-01-24 15:32:09 Relaying wtx 94afa233b9b924eb82d37900754770a671e6f584237fd56c0ee135c434bca94b\r\n2017-01-24 15:32:09 Relaying wtx 6ff221aee6174acc2263c64b1deec94ddfa1ad0b2877b30ccd580476f08ab7e6\r\n2017-01-24 15:32:09 Relaying wtx f43459559a1467dce61266de8c9ff24301f5d47458936f5ce297b3e570781a42\r\n2017-01-24 15:32:09 Relaying wtx fd3b229bd87357ceb521f8657104f0954563d2d80340145a6eeb1766821b1e01\r\n2017-01-24 15:32:09 Relaying wtx a6965943dce6112cdabe9a9e36d39f432b419054dfc893487b5224d7322b61a7\r\n2017-01-24 15:32:09 Relaying wtx 3bc722f7b210a80efa5a8de030f64ab8ce1940a5e45801b57d9444dbbb99e67c\r\n2017-01-24 15:32:09 Relaying wtx 46d82330d53afa5087470118cdf4cf69771aed94851bd9282f107cd383f7676c\r\n--\r\n2017-01-24 15:49:29 receive version message: /bitcoinj:0.14.3/Bitcoin Wallet:5.04/: version 70001, blocks=449806, us=127.0.0.1:8333, peer=274\r\n2017-01-24 15:49:37 UpdateTip: new best=0000000000000000008ce84b11a212f3950dc0d6f1ecccd66361fa2494e73986 height=449807 version=0x20000000 log2_work=85.869022 tx=190354204 date='2017-01-24 15:48:52' progress=1.000000 cache=118.1MiB(57563tx) warning='5 of last 100 blocks have unexpected version'\r\n2017-01-24 15:49:37 AddToWallet a4bdc3db078294e2cf8d935327a40491a7b99d497d473aae9305f346eab76144  update\r\n2017-01-24 15:49:37 AddToWallet 3ed223c549577969cfba7663a324b5642770801303ef140630b8ac1195c7feec  update\r\n2017-01-24 15:49:37 AddToWallet 662b1b2f66179e8be0a055ecaa8728fa069a6974b3b74cfbc87627b7bf50eb76  update\r\n2017-01-24 15:49:37 AddToWallet 6ff221aee6174acc2263c64b1deec94ddfa1ad0b2877b30ccd580476f08ab7e6  update\r\n2017-01-24 15:49:37 AddToWallet b66a224291f1d4ac02ac0d80a67fd4678bff120a2f6b9aeaecffc0ff3a5c0cec  update\r\n2017-01-24 15:49:37 AddToWallet 739688a1de281ef97ceded7a266014f3de656ebacad192b4fe686a95293e9a96  update\r\n2017-01-24 15:49:37 AddToWallet 42138db9663524f808b087725865536e14cff0504badaea59e68c8e9ef0d0521  update\r\n2017-01-24 15:49:37 receive version message: /bitcoinj:0.13.3/Bitcoin Wallet:4.42/: version 70001, blocks=449667, us=127.0.0.1:8333, peer=275\r\n2017-01-24 15:49:37 receive version message: /Bither1.6.0/: version 70001, blocks=449806, us=127.0.0.1:8333, peer=273\r\n```\r\n\r\n`bitcoin-cli gettransaction 6ff221aee6174acc2263c64b1deec94ddfa1ad0b2877b30ccd580476f08ab7e6` gives\r\n\r\n```\r\n{\r\n  \"amount\": -0.99970000,\r\n  \"fee\": -0.00015778,\r\n  \"confirmations\": 24,\r\n  \"blockhash\": \"0000000000000000008ce84b11a212f3950dc0d6f1ecccd66361fa2494e73986\",\r\n  \"blockindex\": 1176,\r\n  \"blocktime\": 1485272932,\r\n  \"txid\": \"6ff221aee6174acc2263c64b1deec94ddfa1ad0b2877b30ccd580476f08ab7e6\",\r\n  \"walletconflicts\": [\r\n  ],\r\n  \"time\": 1485214510,\r\n  \"timereceived\": 1485214510,\r\n  \"bip125-replaceable\": \"no\",\r\n  \"details\": [\r\n    {\r\n      \"account\": \"\",\r\n      \"address\": \"1M5r2SLSuwZYtoR1TA1sM3cmnbhk8Z58wn\",\r\n      \"category\": \"send\",\r\n      \"amount\": -0.99970000,\r\n      \"vout\": 0,\r\n      \"fee\": -0.00015778,\r\n      \"abandoned\": false\r\n    }\r\n  ],\r\n  \"hex\": \"01000000018add02da742785104a5f72d5a23676c0b3e1099c490438b0de66214fa6e000f6010000006b483045022100fd5cac4f9f453aab0b1f36e84055ccdf464ecd65d35d693fcc6bb5a5e2dbd1d102205aeadaabeb0575b4b25281568de960585e26730baec3e574710fafa5bd4dead6012102f963835d3115b83ede335b2c3386682c08c05d43c541f2366c0eb9b201c57bf9feffffff02d06bf505000000001976a914dc4d1a749f9801748b94a0f9890b88e9b0dfbf9488ac9c806986010000001976a91471655990809fdc834a331322a496b74fb763b9bf88acb0dc0600\"\r\n}\r\n```\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n",
      "created_at" : "2017-01-24T19:25:46Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274909599",
      "id" : 274909599,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T19:25:46Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274909599",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "> do you have a super small maxmempool or so?\r\n\r\nNope. I'm running with a pretty standard configuration:\r\n\r\n```\r\nserver=1\r\nrpcuser=bitcoinrpc\r\nrpcpassword=xxxxxxxxx\r\nmaxuploadtarget=144\r\ntxconfirmtarget=6\r\ntxindex=1\r\nkeypool=10000\r\n```\r\n\r\nThat's it",
      "created_at" : "2017-01-24T19:27:34Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274910243",
      "id" : 274910243,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T19:27:34Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274910243",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "What about `grep \"rebroadcast\" | grep \"unconfirmed transactions\"`?",
      "created_at" : "2017-01-24T19:55:12Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274919594",
      "id" : 274919594,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T19:55:12Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274919594",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/649246?v=3",
         "events_url" : "https://api.github.com/users/TheBlueMatt/events{/privacy}",
         "followers_url" : "https://api.github.com/users/TheBlueMatt/followers",
         "following_url" : "https://api.github.com/users/TheBlueMatt/following{/other_user}",
         "gists_url" : "https://api.github.com/users/TheBlueMatt/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/TheBlueMatt",
         "id" : 649246,
         "login" : "TheBlueMatt",
         "organizations_url" : "https://api.github.com/users/TheBlueMatt/orgs",
         "received_events_url" : "https://api.github.com/users/TheBlueMatt/received_events",
         "repos_url" : "https://api.github.com/users/TheBlueMatt/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/TheBlueMatt/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/TheBlueMatt/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/TheBlueMatt"
      }
   },
   {
      "body" : "Yeah, I have a lot of that :P\r\n\r\n\r\n```\r\n2017-01-24 13:47:12 ResendWalletTransactions: rebroadcast 82 unconfirmed transactions\r\n2017-01-24 14:16:16 ResendWalletTransactions: rebroadcast 91 unconfirmed transactions\r\n2017-01-24 15:32:09 ResendWalletTransactions: rebroadcast 72 unconfirmed transactions\r\n2017-01-24 15:53:49 ResendWalletTransactions: rebroadcast 73 unconfirmed transactions\r\n2017-01-24 15:54:33 ResendWalletTransactions: rebroadcast 71 unconfirmed transactions\r\n2017-01-24 16:15:23 ResendWalletTransactions: rebroadcast 73 unconfirmed transactions\r\n2017-01-24 16:42:48 ResendWalletTransactions: rebroadcast 79 unconfirmed transactions\r\n2017-01-24 16:54:58 ResendWalletTransactions: rebroadcast 79 unconfirmed transactions\r\n2017-01-24 17:21:37 ResendWalletTransactions: rebroadcast 100 unconfirmed transactions\r\n2017-01-24 17:32:56 ResendWalletTransactions: rebroadcast 99 unconfirmed transactions\r\n2017-01-24 17:46:12 ResendWalletTransactions: rebroadcast 91 unconfirmed transactions\r\n2017-01-24 18:07:19 ResendWalletTransactions: rebroadcast 86 unconfirmed transactions\r\n2017-01-24 18:47:32 ResendWalletTransactions: rebroadcast 107 unconfirmed transactions\r\n2017-01-24 19:16:41 ResendWalletTransactions: rebroadcast 108 unconfirmed transactions\r\n2017-01-24 19:35:22 ResendWalletTransactions: rebroadcast 92 unconfirmed transactions\r\n2017-01-24 19:44:27 ResendWalletTransactions: rebroadcast 91 unconfirmed transactions\r\n```\r\n\r\nI think the vast majority of them are incoming transactions, accompanied by a support ticket wondering why bitcoin isn't working for them :( ",
      "created_at" : "2017-01-24T20:00:50Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274921340",
      "id" : 274921340,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T20:12:54Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274921340",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "What looks like happened here is the `6ff221a` transaction could not be accepted into your mempool at the time you generated it (but it did make it into your wallet).  This leads to a known sub-optimal behavior where it still counts as spending all the inputs it consumes but does not count as crediting your balance with any of its outputs regardless of whether they are change or otherwise to you.  This explains why your balance \"disappeared\".    \r\n\r\nWhat doesn't make sense is why this situation persisted until a restart?  It should have been the case that your node attempted to reaccept that transaction to your mempool roughly every 30 mins.  And the most likely guess is it had too many unconfirmed parents to be accepted originally, so as soon as few of them were confirmed in blocks, it should have been accepted into the mempool.  However it seems to have not made it back into your mempool until you restarted your node.  I haven't been able to come up with a theory on why that might have been the case.\r\n\r\n",
      "created_at" : "2017-01-24T20:56:41Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-274935875",
      "id" : 274935875,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-24T20:56:41Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/274935875",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/4360349?v=3",
         "events_url" : "https://api.github.com/users/morcos/events{/privacy}",
         "followers_url" : "https://api.github.com/users/morcos/followers",
         "following_url" : "https://api.github.com/users/morcos/following{/other_user}",
         "gists_url" : "https://api.github.com/users/morcos/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/morcos",
         "id" : 4360349,
         "login" : "morcos",
         "organizations_url" : "https://api.github.com/users/morcos/orgs",
         "received_events_url" : "https://api.github.com/users/morcos/received_events",
         "repos_url" : "https://api.github.com/users/morcos/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/morcos/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/morcos/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/morcos"
      }
   },
   {
      "body" : "Theory: Actually running 0.13.1 not 0.13.2? ",
      "created_at" : "2017-01-29T03:10:29Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-275890726",
      "id" : 275890726,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-29T03:10:29Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/275890726",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/858454?v=3",
         "events_url" : "https://api.github.com/users/gmaxwell/events{/privacy}",
         "followers_url" : "https://api.github.com/users/gmaxwell/followers",
         "following_url" : "https://api.github.com/users/gmaxwell/following{/other_user}",
         "gists_url" : "https://api.github.com/users/gmaxwell/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/gmaxwell",
         "id" : 858454,
         "login" : "gmaxwell",
         "organizations_url" : "https://api.github.com/users/gmaxwell/orgs",
         "received_events_url" : "https://api.github.com/users/gmaxwell/received_events",
         "repos_url" : "https://api.github.com/users/gmaxwell/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/gmaxwell/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/gmaxwell/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/gmaxwell"
      }
   },
   {
      "body" : "```\r\n$ bitcoin-cli getinfo\r\n{\r\n  \"version\": 130200,\r\n  \"protocolversion\": 70015,\r\n  \"walletversion\": 60000,\r\n```\r\n\r\nIt's possible that when I upgraded to 0.13.2 I forgot to restart my client, but I very much strongly doubt it. I also recall checking to make sure I was on the latest, but I'm not 100%.\r\n\r\nWas there an issue in 0.13.1 that could've caused it?\r\n\r\n",
      "created_at" : "2017-01-29T03:13:30Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-275890825",
      "id" : 275890825,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-29T03:13:30Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/275890825",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   },
   {
      "body" : "0.13.1 doesn't reattempt the mempool insertion, this was fixed in 0.13.2.  Any chance you have enough debug logs to confirm one way or the other? (the logs log the version at start).\r\n\r\n(If you were on 0.13.1 this would be a known issue that was fixed in 0.13.2)",
      "created_at" : "2017-01-29T03:18:15Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-275891000",
      "id" : 275891000,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-29T03:18:58Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/275891000",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/858454?v=3",
         "events_url" : "https://api.github.com/users/gmaxwell/events{/privacy}",
         "followers_url" : "https://api.github.com/users/gmaxwell/followers",
         "following_url" : "https://api.github.com/users/gmaxwell/following{/other_user}",
         "gists_url" : "https://api.github.com/users/gmaxwell/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/gmaxwell",
         "id" : 858454,
         "login" : "gmaxwell",
         "organizations_url" : "https://api.github.com/users/gmaxwell/orgs",
         "received_events_url" : "https://api.github.com/users/gmaxwell/received_events",
         "repos_url" : "https://api.github.com/users/gmaxwell/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/gmaxwell/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/gmaxwell/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/gmaxwell"
      }
   },
   {
      "body" : "The logs don't go that far back, unfortunately. Since there's not probably not an awful lot here for people to work with, I'll just close the issue and try reproduce it (I've now got code that keeps track what my wallet balance should be and if it diverges from what bitcoin reports, I'll know the issue has resurfaced)",
      "created_at" : "2017-01-29T03:31:08Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/9620#issuecomment-275891467",
      "id" : 275891467,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/9620",
      "updated_at" : "2017-01-29T03:31:08Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/275891467",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/9326759?v=3",
         "events_url" : "https://api.github.com/users/RHavar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/RHavar/followers",
         "following_url" : "https://api.github.com/users/RHavar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/RHavar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/RHavar",
         "id" : 9326759,
         "login" : "RHavar",
         "organizations_url" : "https://api.github.com/users/RHavar/orgs",
         "received_events_url" : "https://api.github.com/users/RHavar/received_events",
         "repos_url" : "https://api.github.com/users/RHavar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/RHavar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/RHavar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/RHavar"
      }
   }
]
